tracer/src/Datadog.Trace/Ci/CIVisibility.cs

```diff
@@ -68,7 +69,25 @@ public static void Initialize()
             var eventPlatformProxyEnabled = false;
             if (!_settings.Agentless)
             {
-                discoveryService = DiscoveryService.Create(new ImmutableExporterSettings(_settings.TracerSettings.Exporter));
+                if (!_settings.ForceAgentsEvpProxy)
+                {
+                    discoveryService = new DiscoveryService(
+                        AgentTransportStrategy.Get(
+                            new ImmutableExporterSettings(_settings.TracerSettings.Exporter),
+                            productName: "discovery",
+                            tcpTimeout: TimeSpan.FromSeconds(5),
+                            AgentHttpHeaderNames.MinimalHeaders,
+                            () => new MinimalAgentHeaderHelper(),
+                            uri => uri),
```

Comment:
This is essentially the same as the existing transport strategy, just with a smaller tcp timeout right? I'd be inclined to make this the default, expose it as a helper, and just share everywhere (i.e. CI visibility + normal)